src/consensus/coin.h

```diff
@@ -101,6 +101,10 @@ class Coin {
         return value_ > another.value_;
     }
 
+    explicit operator bool() const {
```

Comment:
Although we use `explicit` here, overloading operator bool is not so good. We can try other method name like `IsZero`

Reply:
Why not? I think Coin is basically a mimic of uint64

Reply:
Yes but casting primitive type to bool is from C because C does not have boolean type. In my opinion, `Coin` is indeed a user defined class with limited operations compared to `uint64`. Casting to bool reduces readability.

Reply:
I'll just use GetValue then